SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -75,6 +78,38 @@ def upgrade():
     pass
 
 
+# returns a JSON dict {repo_id: proxy}
+def get_proxies(session, args):
+    config = ConfigParser.ConfigParser({'proxy': '_none_'})
+    config.read('/etc/yum.repos.d/xcp-ng.repo')
+    new_dict = dict((section, config.get(section, 'proxy', '_none_')) for section in config.sections())
+    return json.dumps(new_dict)
+
+
+# expects a JSON dict in a 'proxies' argument. The dict should be of the form {repo_id: proxy} with the special proxy
+# '_none_' used for removal.
+# example: {"xcp-ng-base": "http://192.168.100.82:3142"}
+def set_proxies(session, args):
+    # '{"xcp-ng-base": "http://192.168.100.82:3142"}'
+    # '{"xcp-ng-base": "_none_"}'
+    special_url_prefix = 'http://HTTPS///'
+    https_url_prefix = 'https://'
+    proxies = json.loads(args['proxies'])
+    config = ConfigParser.ConfigParser()
+    config.read('/etc/yum.repos.d/xcp-ng.repo')
+    for section in proxies:
+        if config.has_section(section):
+            config.set(section, 'proxy', proxies[section])
+            url = config.get(section, 'baseurl')
+            if proxies[section] == '_none_' and url.startswith(special_url_prefix):
+                config.set(section, 'baseurl', https_url_prefix + url[len(special_url_prefix):])
+            elif proxies[section] != '_none_' and url.startswith(https_url_prefix):
```

Comment:
Would be good to add a "else" and return a warning or error if it failed.

Also, we might want to **not** write the file at all if anything doesn't go as expected.

Reply:
I feel like if we have a complicated behavior we'll need a complicated error reporting and handling, a good communication with the front side, then to the user. You have to balance that with an API that say readFile()/writeFile(), and the frontside user deciding everything and reporting what they want to the user might that might be be simpler (and they can even show the complete file for user validation before accepting the change).

Here we're putting some logic in the server, some logic in the front and now we have to explain why stuff went wrong to the users.

Reply:
With logic on the server, we can have it evolve without modifying the client (XO), if we think it may change over time. Else maybe it's better indeed to use such a generic read/write API, if it does not raise security issues because it's too powerful.

I suppose there's a design discussion to be held with @julien-f and maybe @olivierlambert before we go on then.

Reply:
I think there is a very low probability that the plugin code would change without also changing something on the XO side. And also XO has a faster schedule. 

But Oliver was not too keen on adding a generic "do something on the server as root" plugin.